lang/rust/avro/src/codec.rs

```diff
@@ -82,18 +82,16 @@ impl Codec {
             }
             #[cfg(feature = "snappy")]
             Codec::Snappy => {
-                use byteorder::ByteOrder;
-
                 let mut encoded: Vec<u8> = vec![0; snap::raw::max_compress_len(stream.len())];
                 let compressed_size = snap::raw::Encoder::new()
                     .compress(&stream[..], &mut encoded[..])
                     .map_err(Error::SnappyCompress)?;
+                encoded.truncate(compressed_size);
```

Comment:
As we know we will add u32 checksum, wouldn't it be more efficient with
`encoded.truncate(compressed_size + 4);`

Reply:
Good suggestion! Thanks!